LandSandBoat/server
# Refactor several status effect functions

**_I affirm:_**

----

## What does this pull request do?

Refactors several status effect functions

### To add and remove a single bit to the mask:

#### Before
- `setFlag() & ENUM`
- `unsetFlag() & ENUM`

#### After
- `addEffectFlag(ENUM)`
- `delEffectFlag(ENUM)`

### We now have:
- `getEffectFlags()`
- `setEffectFlags(fullmask)`
- `addEffectFlag(flagBit)`
- `delEffectFlag(flagBit)`
- `hasEffectFlag(flagBit)`

#### `flags` plural deal with the full mask. `flag` singular deal with a single enumerated bit.

#### And I have renamed some other functions make them less likely to be confused with identically named functions in the future:
- `getType()` -> `getEffectType()`
- `getSlot()` -> `getEffectSlot()`
- `setType()` -> `setEffectType()`
- `setSlot()` -> `setEffectSlot()`
- `setName()` -> `setEffectName()`

----

## Steps to test these changes
_I suggest rigging yourself a little GM command or 2 instead of all the `!exec` I did_
1. get an effect (any effect) in script: `local status = entity:getStatusEffect(xi.effects.EFFECT_ID_HERE)`
2. test getting its full mask, `print(status:getEffectFlags())`
3. setting its full mask, `status:setEffectFlags(0)`
4. get them again to verify it changed,
5. adding a bit, `status:addEffectFlag(1)`
6. checking it has the bit `if status:hasEffectFlag(1) then print("success!") end`
7. deleting the bit `status:delEffectFlag(1)`
8. confirm bit was deleted `if NOT status:hasEffectFlag(1) then print("success!") end`
9. confirm existing content using these features still works
examples (that I checked): 
- monsters scripts granting spikes set the mask to use the death flag only so that it can only be removed by death, 
- corsair shot Job Ability scripts get the effect type
- Abyssea visitant
- ATMA
- PUP maneuvers
- THF's Larceny and Aura Steal
